client/reducers/scenario.js

```diff
@@ -11,21 +25,32 @@ const initialState = {
     status: 1
 };
 
-export default function(state = initialState, action) {
-    switch (action.type) {
-        case SET_SCENARIO:
-        case SET_SCENARIOS: {
-            return {
-                ...state,
-                ...action.payload
-            };
-        }
-        case SET_SLIDES:
-            return {
-                ...state,
-                ...action.payload
-            };
-        default:
-            return state;
+export const scenario = (state = initialState, action) => {
+    const { scenario, slides, type } = action;
+
+    if (type === SET_SCENARIO || type === GET_SCENARIO_SUCCESS) {
+        return {
+            ...state,
+            ...scenario
+        };
+    }
+
+    if (type === SET_SLIDES) {
+        return {
+            ...state,
+            ...slides
+        };
+    }
+
+    return state;
+};
+
+export const scenarios = (state = [], action) => {
+    const { scenarios, type } = action;
+
+    if (type === SET_SCENARIOS || type === GET_SCENARIOS_SUCCESS) {
+        return scenarios;
```

Comment:
We don't need to include the original state information? Like we do on line 40 for example?

Reply:
Also is anything using the `GET_SCENARIOS_SUCCESS` reducer at the moment? Or is this set up for later logic?

Reply:
In this case, we don't want to do that because the "original state" might be the previous array of scenarios, which would mean 2, 3, 4... etc versions of the same scenario in list.